crates/sncast/src/starknet_commands/account/deploy.rs

```diff
@@ -98,17 +99,33 @@ async fn deploy_from_keystore(
     }
 
     let salt = account_data.get_salt_as_felt()?;
-    let class_hash = account_data.get_class_hash_as_felt()?;
 
     // TODO: Improve code for checking if address exists or remove this logic
     let account_type = account_data.get_account_type()?;
-    let calldata = match account_type {
-        "open_zeppelin" => vec![private_key.verifying_key().scalar()],
-        "argent" => vec![private_key.verifying_key().scalar(), FieldElement::ZERO],
-        _ => panic!("Invalid account type"),
-    };
+    let class_hash = account_data.get_class_hash_as_felt()?;
+    account_data.get_class_hash_as_felt()?;
 
-    let address = get_contract_address(salt, class_hash, &calldata, FieldElement::ZERO);
+    let address = match account_type {
+        AccountType::Argent => get_contract_address(
+            salt,
+            class_hash,
+            &[private_key.verifying_key().scalar(), FieldElement::ZERO],
+            FieldElement::ZERO,
+        ),
+        AccountType::Oz => get_contract_address(
+            salt,
+            class_hash,
+            &[private_key.verifying_key().scalar()],
+            chain_id,
+        ),
+        AccountType::Braavos => get_contract_address(
+            salt,
+            parse_number(BRAAVOS_BASE_ACCOUNT_CLASS_HASH)
+                .expect("Failed to parse Braavos account class hash"),
+            &[private_key.verifying_key().scalar()],
+            chain_id,
+        ),
+    };
```

Comment:
I think the function `update_keystore_account` needs to be potentially modified for braavos as well. There is an additional key field `context` that is removed by `starkli` after deployment. Could you check this?